compiler/frontend/src/mir_optimize/constant_folding.rs

```diff
@@ -70,6 +76,11 @@ pub fn fold_constants(
     *expression = evaluated_call;
 }
 
+enum BuiltinResult {
+    Returns(Expression),
+    Panics(String),
+}
```

Comment:
We could also `impl From<Expression> for Option<BuiltinResult>` to make `run_builtin(…)` more concise

Reply:
If I understand your plan correctly, then we'd have lots of `expr.into()` instead of `Returns(expr)` in the `run_builtin`? In that case, I prefer the explicit `Returns` one.

Reply:
`Returns(expr).into()` would become `expr.into()`, but we can also leave this as is